src/lib_ccx/ccx_encoders_helpers.c

```diff
@@ -468,6 +540,6 @@ void shell_sort(void *base, int nb, size_t size, int(*compar)(const void*p1, con
 
 void ccx_encoders_helpers_perform_shellsort_words(void)
 {
-	shell_sort(spell_lower, spell_words, sizeof(*spell_lower), string_cmp_function, NULL);
-	shell_sort(spell_correct, spell_words, sizeof(*spell_correct), string_cmp_function, NULL);
+	shell_sort(spell_lower.words,   spell_lower.len,   sizeof(*spell_lower.words),   string_cmp_function, NULL);
```

Comment:
Seems strange to be passing  a length there and also using sizeof, why?

Reply:
I didn't write that code but:
> Seems strange to be passing a length

So that it the function knows how many elements it needs

> using sizeof

To know the size of each element

The best would probably to use a stdlib sorting function.

Also I didn't sort the list of profane words, which means binary search won't work (so far it has worked because the list was already sorted).

I'll be fixing that tomorrow

Reply:
The use of shell sort was added in #41 because it was faster than quicksort.

Also to clarify, `spell_words` was the length of the array (now `spell_correct.len`) so no behaviour should have changed.

Reply:
That's not too reassuring :-) You need to test a lot and be sure that you are not introducing any bug...